### PR TITLE
Single release endpoint

### PR DESCRIPTION
**Pull request by agduncan94**
Now only handle the push webhook (before was install and release endpoints). It simplifies the code a lot.

https://github.com/dockstore/dockstore/issues/2990

### CODE REVIEW

File: upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js [hunk 1]
```diff
@@ -110,59 +82,42 @@ function processEvent(event, callback) {
...
-                        callback(null);
-                    } else {
-                        // Server error, retry
-                        console.info('Retrying call');
-                        callback({ "statusCode": response.statusCode, "statusMessage": response.statusMessage });
-                    }
-                });   
+    var path = process.env.API_URL;
+
+    // A push has been made for some repository (ignore pushes that are deletes)
```
Comment (coverbeck) [inline upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js]: What exactly is a delete event and why are we ignoring it? Does it mean if somebody deletes a branch, the branch will continue to exist on Dockstore?

Comment (agduncan94) [reply upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js]: Yes exactly. We haven't decided how we would handle deletes yet so for now this is ignored.

Comment (coverbeck) [reply upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js]: This seems like a departure from existing behavior, doesn't it?

Comment (agduncan94) [reply upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js]: I thought that if a branch is deleted on GH it stays on Dockstore. Or maybe that is for tools only.

Comment (coverbeck) [reply upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js]: I think it deletes them: https://github.com/dockstore/dockstore/blob/853fb797990ec557cf574f6d5e28d36acad5aff8/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java#L164

Comment (denis-yuen) [reply upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js]: Huh. I wonder if that works. I wonder if I wrote it.

Comment (Ldcabansay) [reply upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js]: For a branch, it deletes them, at least thats what happened when I was checking out the hotfix 1.8.2 behavior. I didn't check what happens if a tag was deleted though...

Comment (agduncan94) [reply upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js]: @denis-yuen should a delete branch request from GitHub also delete the branch on Dockstore?

Comment (denis-yuen) [reply upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js]: Yeah, I think if that's the current behavior we should match it.

Comment (agduncan94) [reply upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js]: Ok I'll confirm that that is what we currently do and if it is will add this feature.

Comment (agduncan94) [reply upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js]: Looks like that is current behaviour, will update.

Comment (agduncan94) [reply upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js]: Sorry If it looked like I ignored people I swear GitHub didn't load any new messages from the link Charles sent onwards.

**Comment by agduncan94** [comment]
Going to add the delete endpoint in an upcoming PR.

**Pull request merged**